modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/generalized_plane_strain.md

```diff
@@ -0,0 +1,125 @@
+
+A further condition is required associated with the out-of-plane direction. If a constant strain is prescribed as the deformation compatibility condition in the $z$-direction
+\begin{equation}
+\epsilon_{zz} = \bar{\epsilon}_{zz}
+\end{equation}
+Alternatively, a force as the stress resultant $\bar{N}_{z}$ in the $z$-direction can be prescribed, the condition is the equilibrium condition in $z$-direction, given as
+\begin{equation}
+N_{z} = \int_{A}{\sigma_{zz}dA} = \bar{N}_{z}
+\end{equation}
+The stress resultant $N_{z}$ conjugates with constant strain $\epsilon_{zz}$.
+
+The formulation above for generalized plane strain problem shares many similarities with that of the conventional plane stress or plane strain problem. All the differences are associated with the introduction of an additional degree of freedom for the out-of-plane direction.
+
+## Implementation and Usage
+
+The out-of-plane strain is a scalar variable, and it can be added to the standard system of equations for a mechanics problem, where $\bf{u}_x$ and $\bf{u}_y$ represent
+the displacement vectors in the $x$ and $y$ directions, $\bf{f}_x$ and $\bf{f}_y$ represent the corresponding reaction forces. The partitioned linearized system of equations, in which the block entries in the stiffness matrix are represented by subscripted $\bf{K}$ terms, can be written including the scalar strain variable as follows:
+\begin{equation}
+\left[
+\begin{array}{cc|c}
+\bf{K}_{xx} & \bf{K}_{xy} & \vdots \\
+\bf{K}_{yx} & \bf{K}_{yy} & \vdots \\ \hline
+\cdots & \cdots & K_{zz}
+\end{array}
+\right]
+\left\{
+\begin{array}{c}
+\bf{u}_x\\
+\bf{u}_y\\ \hline
+\epsilon_{zz}
+\end{array}
+\right\}
+=
+\left\{
+\begin{array}{c}
+\bf{f}_x\\
+\bf{f}_y\\ \hline
+N_{z}
+\end{array}
+\right\}
+\end{equation}
+
+The off-diagonal entries are nonzero, but not shown here.
+
+### List of MOOSE objects used in the implementation
+
+Objects available for generalized plane strain:
+- [Stress Divergence Kernel](/StressDivergence.md): in-plane equilibrium equation
+- [Stress Models](tensor_mechanics/Stresses.md): full stress tensor calculation
+
+Objects specific for generalized plane strain:
+- [Generalized Plane Strain ScalarKernel](/GeneralizedPlaneStrain.md): out-of-plane equilibrium condition
+- [Generalized Plane Strain UserObject](/GeneralizedPlaneStrainUserObject.md): residual and diagonal Jacobian calculation for scalar out-of-plane strain variable
+- [Generalized Plane Strain Off-diagonal Kernel](/GeneralizedPlaneStrainOffDiag.md): in-plane displacement variables and scalar out-of-plane strain coupling
+- [Strain Calculations](tensor_mechanics/Strains.md): in-plane strain calculation and formation of full strain tensor considering the scalar out-of-plane strain
+
+### How to use generalized plane strain model
```

Comment:
Make this a second level (`##`) header--it could be the part that most readers will care about the most!